packages/permission-controller/src/Caveat.ts

```diff
@@ -125,6 +182,17 @@ export type CaveatSpecificationBase = {
   // TODO: Replace `any` with type
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   validator?: CaveatValidator<any>;
+
+  /**
+   * The merger function used to merge caveats of the associated type during
+   * incremental permission requests. The values must be merged in the fashion
+   * of a right-biased union.
+   *
+   * @see `ARCHITECTURE.md` for more details.
+   */
+  // TODO: Replace `any` with type
```

Comment:
Can we do this now? :)

Reply:
Unfortunately, I don't think that's a good idea. I'm not even sure what the problem is, but my current best guess is that it requires genericizing `CaveatSpecificationBase`. A quick stab at doing that indicates that this would clutter this PR even more. I can potentially take a stab at it afterwards, though.